addons/website_payment/static/src/snippets/s_donation/donation_snippet.js

```diff
@@ -0,0 +1,167 @@
+    };
+
+    setup() {
+        this.currency = null;
+        this.activeButton = null;
+        this.rangeSliderEl = this.el.querySelector('#s_donation_range_slider');
+        this.defaultAmount = this.el.dataset.defaultAmount;
+        if (!!this.rangeSliderEl) {
+            this.rangeSliderEl.value = this.defaultAmount;
+            this.setBubble(this.rangeSliderEl);
+        }
+    }
+
+    async willStart() {
+        cachedCurrency = cachedCurrency || this.waitFor(rpc("/website/get_current_currency"));
+        this.currency = await cachedCurrency;
+    }
+
+    start() {
+        const currencyEls = this.el.querySelectorAll('.s_donation_currency');
+        for (const currencyEl of currencyEls) {
+            currencyEl.remove();
+        }
+
+        const prefilledButtonEls = this.el.querySelectorAll('.s_donation_btn, .s_range_bubble');
+        for (const prefilledButtonEl of prefilledButtonEls) {
+            const before = result.position === "before";
+            const currencySymbol = document.createElement('span');
+            currencySymbol.innerText = result.symbol;
+            currencySymbol.classList.add('s_donation_currency', before ? "pe-1" : "ps-1");
+            this.insert(currencySymbol, prefilledButtonEl, before ? "afterbegin" : "beforeend");
+        }
+
+        const customButtonEl = this.el.querySelector("#s_donation_amount_input");
+        if (customButtonEl) {
+            this.registerCleanup(() => { customButtonEl.style.maxWidth = "" });
+            const canvasEl = document.createElement("canvas");
+            const context = canvasEl.getContext("2d");
+            context.font = window.getComputedStyle(customButtonEl).font;
+            const width = context.measureText(customButtonEl.placeholder).width;
+            customButtonEl.style.maxWidth = `${Math.ceil(width) + CUSTOM_BUTTON_EXTRA_WIDTH}px`;
+        }
+    }
+
+    /**
+     * @param {HTMLInputElement} rangeEl
+     */
+    setBubble(rangeEl) {
+        const bubble = this.el.querySelector('.s_range_bubble');
+        const val = rangeEl.value;
+        const min = rangeEl.min || 0;
+        const max = rangeEl.max || 100;
+        const newVal = Number(((val - min) * 100) / (max - min));
+        const tipOffsetLow = 8 - (newVal * 0.16); // the range thumb size is 16px*16px. The '8' and the '0.16' are related to that 16px (50% and 1% of 16px)
+
+        for (let child of bubble.childNodes) {
+            if (child.nodeType === 3) {
+                child.nodeValue = val;
+            }
+        }
+        // Sorta magic numbers based on size of the native UI thumb (source: https://css-tricks.com/value-bubbles-for-range-inputs/)
+        bubble.style.left = `calc(${newVal}% + (${tipOffsetLow}px))`;
+    }
+
+    /**
+     * @param {Event} ev
+     */
+    onClickPrefilled(ev) {
+        this.activeButton = ev.currentTarget;
+        this.activeButton.classList.add("active");
```

Comment:
This should not be needed anymore, is it? (Also actually: `this.activeButtonEl`)